src/GraphQl/Resolver/Factory/ResolverFactory.php

```diff
@@ -32,17 +33,19 @@ public function __construct(
     public function __invoke(string $resourceClass = null, string $rootClass = null, Operation $operation = null): callable
     {
         return function (?array $source, array $args, $context, ResolveInfo $info) use ($resourceClass, $rootClass, $operation) {
+            // special treatment for nested resources without a resolver/provider
+            if (\array_key_exists($info->fieldName, $source ?? []) && $operation instanceof Query && $operation->getNested() && !$operation->getResolver() && !$operation->getProvider()) {
+                $body = $source[$info->fieldName];
+
+                return \is_array($body) ? $this->resolve($source, $args, $info, $rootClass, $operation, new ArrayPaginator($body, 0, \count($body))) : $body;
+            }
+
             // Data already fetched and normalized (field or nested resource)
             if ($body = $source[$info->fieldName] ?? null) {
-                // special treatment for nested resources without a resolver/provider
-                if ($operation instanceof Query && $operation->getNested() && !$operation->getResolver() && !$operation->getProvider()) {
-                    return $this->resolve($source, $args, $info, $rootClass, $operation, new ArrayPaginator($body, 0, \count($body)));
-                }
-
                 return $body;
             }
 
-            if (null === $resourceClass && \array_key_exists($info->fieldName, $source ?? [])) {
+            if ((null === $resourceClass || !is_iterable($body)) && \array_key_exists($info->fieldName, $source ?? [])) {
```

Comment:
Meanwhile I found a working solution: this tiny change fixes the issue with `null` values.

Reply:
shouldn't we check the `PropertyMetadata` type for this?

Reply:
refactored